File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        n = len(nums)

        mp = {}

        for i in nums:
            mp[i] = 1 + mp.get(i,0)

        freq = [[] for i in range(n)]
        for num, count in mp.items():
            freq[count-1].append(num)
    
        res=[]
        print(mp)
        print(freq)
        for i in range(n-1,-1,-1):
            if k==0:
                break
            for j in freq[i]:
                k=k-1
                res.append(j)
                if k==0:
                    break
        return res

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-8.java
class Solution {
    public int findMin(int[] nums) {
        int n=nums.length;
        int l=0, h=n-1;
        while(l<=h)
        {
            int m=(l+h)/2;

            if(m>0 && nums[m-1]>nums[m])
            {
                return nums[m];

            }
            else if(nums[m]<nums[h])
                h=m-1;
            else
                l=m+1;

        }
        return nums[0];
    }
}
